stubs/docopt/docopt.pyi

```diff
@@ -0,0 +1,11 @@
+from typing import Any, Optional, Union
+
+__version__: str
+
+def docopt(
+    doc: str,
+    argv: Optional[list[str]] = ...,
```

Comment:
Any iterable works:

```suggestion
    argv: Optional[Iterable[str]] = ...,
```

You will have to import `Iterable` from `typing`.

Reply:
Actually, it can iterate twice in `extras()` function, so iterators won't work correctly. Ideally there would be some way to say "iterable but not iterator", but there isn't, so we can do this instead:

```suggestion
    argv: Optional[Sequence[str]] = ...,
```

Again, you need to import `Sequence` from `typing`.

Reply:
Ok, one more attempt. I have now understood the code wrongly at least two times. I was confused by how `TokenStream`'s `__init__` overrides `list`'s `__init__`.

Actually, `args` can be `None` or a string (technically, anything with `.split()`) or an iterable. `None` is handled in `docopt()` and the other two choices in `TokenStream.__init__`.

```suggestion
    argv: Union[None, str, Iterable[str]] = ...,
```

Reply:
I changed it to that. And I assume I should use `Iterable` from `typing` as opposed to from `collections.abc` because this stub supports python 2 as well?

Reply:
Both work. I'll accept it either way. But please also add support for passing in a string.

Reply:
@Akuli  Oops, I didn't see your multiple follow ups, that's why I was being oblivious in those other comments.